File: backend/Helpdesk.Api/Models/Ticket.cs
using System.ComponentModel.DataAnnotations;

namespace Helpdesk.Api.Models;

public class Ticket
{
    public int Id { get; set; }

    [Required, MaxLength(200)]
    public string Title { get; set; } = string.Empty;

    [Required, MaxLength(2000)]
    public string Description { get; set; } = string.Empty;

    public TicketCategory Category { get; set; }
    public TicketPriority Priority { get; set; }
    public TicketStatus Status { get; set; } = TicketStatus.Open;

    public DateTime CreatedAt { get; set; }
    public DateTime UpdatedAt { get; set; }

    public List<Comment> Comments { get; set; } = new();
}

File: frontend/helpdesk-ui/src/types/ticket.ts
export type TicketCategory = 0 | 1 | 2; // IT, Facilities, General
export type TicketPriority = 0 | 1 | 2; // Low, Medium, High
export type TicketStatus = 0 | 1 | 2;   // Open, InProgress, Closed

export type CommentItem = {
    id: number;
    ticketId: number;
    author: string;
    message: string;
    createdAt: string;
};

export type TicketListItem = {
    id: number;
    title: string;
    createdBy: string;
    category: TicketCategory;
    priority: TicketPriority;
    status: TicketStatus;
    createdAt: string;
    updatedAt: string;
    commentCount: number;
};

export type TicketDetails = {
    id: number;
    title: string;
    description: string;
    createdBy: string;
    category: TicketCategory;
    priority: TicketPriority;
    status: TicketStatus;
    createdAt: string;
    updatedAt: string;
    comments: CommentItem[];
};
